projects/ion/src/lib/sidebar/utils.ts

```diff
@@ -4,14 +4,18 @@ function selectItem(items: Item[], index: number): void {
   items[index].selected = true;
 }
 
-function callItemAction(items: Item[], index: number): void {
+export function callItemAction(items: Item[], index: number): void {
   if (items[index].action) {
     items[index].action();
   }
 }
 
-export function unselectAllItems(items: Item[]): void {
+export function unselectAllItems(items: Item[], exceptItemIndex = -1): void {
   items.forEach((item) => (item.selected = false));
+
+  if (exceptItemIndex >= 0) {
+    items[exceptItemIndex].selected = true;
+  }
 }
```

Comment:
```suggestion
export function unselectAllItems(items: Item[], exceptItemIndex?: number): void {
  items.forEach((item, index) => {
    item.selected = exceptItemIndex !== undefined ? index === exceptItemIndex : false;
  });
}
```

Can you check this solution?

```ts
import { unselectAllItems } from './unselectAllItems';

describe('unselectAllItems', () => {
  const items = [
    { id: 1, selected: true },
    { id: 2, selected: false },
    { id: 3, selected: true },
  ];

  it('should deselect all items if no index is provided', () => {
    unselectAllItems(items);
    expect(items.every(item => !item.selected)).toBe(true);
  });

  it('should deselect all items except the one at the specified index', () => {
    const exceptItemIndex = 1;
    unselectAllItems(items, exceptItemIndex);
    expect(items[exceptItemIndex].selected).toBe(true);
    expect(items.filter(item => item.selected).length).toBe(1);
  });

  it('should not change the selection if an invalid index is provided', () => {
    const invalidIndex = 10;
    unselectAllItems(items, invalidIndex);
    expect(items.every(item => item.selected)).toBe(true);
  });
});
```